source/microbit/microbitimage.cpp

```diff
@@ -439,6 +410,39 @@ mp_obj_t microbit_image_fill(mp_obj_t self_in, mp_obj_t n_in) {
 }
 MP_DEFINE_CONST_FUN_OBJ_2(microbit_image_fill_obj, microbit_image_fill);
 
+mp_obj_t microbit_image_blit(mp_uint_t n_args, const mp_obj_t *args) {
+    microbit_image_obj_t *self = (microbit_image_obj_t*)args[0];
+    check_mutability(self);
+
+    mp_obj_t src = args[1];
+    if (mp_obj_get_type(src) != &microbit_image_type) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_TypeError, "expecting an image"));
+    }
+    if (n_args == 7) {
+        nlr_raise(mp_obj_new_exception_msg(&mp_type_TypeError,
+            "must specify both offsets"));
+    }
+    mp_int_t x = mp_obj_get_int(args[2]);
+    mp_int_t y = mp_obj_get_int(args[3]);
+    mp_int_t w = mp_obj_get_int(args[4]);
+    mp_int_t h = mp_obj_get_int(args[5]);
+    if (w < 0 || h < 0) {
```

Comment:
This is probably a crazy idea, but whenever I see a `ValueError` raised I begin to wonder whether that case could actually make sense and be useful, and in this case I think it would be natural to simply copy the region as mirrored (with negative width) or flipped up-side-down (with negative height), or both. I know this is probably putting too much functionality into a single method, and makes it really complex, but I thought I would just throw it here anyways. Just ignore if you think it's silly.